bluesky/tests/test_flyer.py

```diff
@@ -0,0 +1,319 @@
+import pytest
+
+from bluesky import Msg
+from bluesky.plans import fly, count
+from bluesky.run_engine import IllegalMessageSequence
+from bluesky.tests import requires_ophyd
+from ophyd import Component as Cpt, Device
+from ophyd.sim import NullStatus, TrivialFlyer
+
+
+@requires_ophyd
+def test_flyer_with_collect_asset_documents(RE):
+
+    from ophyd.sim import det, new_trivial_flyer, trivial_flyer
+    from bluesky.preprocessors import fly_during_wrapper
+
+    assert hasattr(new_trivial_flyer, "collect_asset_docs")
+    assert not hasattr(trivial_flyer, "collec_asset_docs")
+    RE(fly_during_wrapper(count([det], num=5), [new_trivial_flyer, trivial_flyer]))
+
+
+@requires_ophyd
+def test_collect_uncollected_and_log_any_errors(RE):
+    # test that if stopping one motor raises an error, we can carry on
+    collected = {}
+
+    from ophyd.sim import TrivialFlyer
+
+    class DummyFlyerWithFlag(TrivialFlyer):
+        def collect(self):
+            collected[self.name] = True
+            super().collect()
+
+    class BrokenDummyFlyerWithFlag(DummyFlyerWithFlag):
+        def collect(self):
+            super().collect()
+            raise Exception
+
+    flyer1 = DummyFlyerWithFlag()
+    flyer1.name = "flyer1"
+    flyer2 = DummyFlyerWithFlag()
```

Comment:
I know this is a test you just moved, but I suspect one of these was meant to be `BrokenDummyFlyerWithFlag`